src/mon/OSDMonitor.cc

```diff
@@ -5072,6 +5072,20 @@ void OSDMonitor::get_pools_health(
           detail->push_back(s);
       }
     }
+
+    if (sum.num_objects > 0 && pool.application_metadata.empty()) {
```

Comment:
actually, no..  i think we should condition this on the mon::feature get_required_features() for LUMINOUS.

And we should automagically enable applications for pools we recognized during mon upgrade...